### PR TITLE
PI-21829 OAuth 2.0 Authentication Filter Implementation

### PR DESCRIPTION
**Pull request by ketanmulay-appdirect**
PI-21829

#### Description
Implement the changes required to authorize the incoming oAuth 2.0 requests to SDK for the below endpoints -
"/api/v2/integration/**"
"/api/v2/domainassociation/**"
"/api/v2/migration/**"
"/api/v2/restrictions/**"

#### Manual merge checklist:
- [ ] Code Review completed
- [ ] Code coverage
- [ ] QA Validation completed
  - Testrail TestCase/Plan link:
- [ ] Approved by a PI tech lead
- [ ] Github checks all green

### CODE REVIEW

**Comment by ketanmulay-appdirect** [comment]
Dev Testing -

image

image

**Review by reshmakadam-appdirect — APPROVED** [review]

**Review by reshmakadam-appdirect — DISMISSED** [review]

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationService.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.appdirect.sdk.web.oauth;
+
+import javax.servlet.Filter;
+
+public class DeveloperSpecificOAuth2AuthorizationService {
```
Comment (abidhkhan) [inline src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationService.java:5]: `DeveloperSpecificOAuth2AuthorizationService` sounds very odd.  Can we just say `OAuth2AuthorizationService`

Comment (abidhkhan) [inline src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationService.java:5]: Should not be `interface` ?

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationSupplier.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import javax.servlet.Filter;
+
+/**
+ * Implementations of this interface provide a way for the service-integration-sdk
+ * to retrieve the oAuth2 filter. Each SDK client application must contain a bean
+ * of this type in its application context in order for the communication with AppMarket to work.
+ */
+@FunctionalInterface
+public interface DeveloperSpecificOAuth2AuthorizationSupplier {
```
Comment (abidhkhan) [inline src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationSupplier.java:11]: `OAuth2AuthorizationSupplier`

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationSupplierImpl.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.appdirect.sdk.web.oauth;
+
+import javax.servlet.Filter;
+
+public class DeveloperSpecificOAuth2AuthorizationSupplierImpl implements DeveloperSpecificOAuth2AuthorizationSupplier{
```
Comment (abidhkhan) [inline src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationSupplierImpl.java:5]: `OAuth2AuthorizationSupplierImpl`

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2FeatureFlagSupplier.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.appdirect.sdk.web.oauth;
+
+/**
+ * Implementations of this interface provide a way for the service-integration-sdk
+ * to check if the oAuth2 authorization is enabled. Each SDK client application must contain a bean
+ * of this type in its application context in order for the communication with AppMarket to work.
+ */
+@FunctionalInterface
+public interface DeveloperSpecificOAuth2FeatureFlagSupplier {
```
Comment (abidhkhan) [inline src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2FeatureFlagSupplier.java:9]: `OAuth2FeatureFlagSupplier`

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2FeatureFlagService.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package com.appdirect.sdk.web.oauth;
+
+public class DeveloperSpecificOAuth2FeatureFlagService {
```
Comment (abidhkhan) [inline src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2FeatureFlagService.java:3]: `OAuth2FeatureFlagService`

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2FeatureFlagSupplierImpl.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package com.appdirect.sdk.web.oauth;
+
+public class DeveloperSpecificOAuth2FeatureFlagSupplierImpl implements DeveloperSpecificOAuth2FeatureFlagSupplier{
```
Comment (abidhkhan) [inline src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2FeatureFlagSupplierImpl.java:3]: `OAuth2FeatureFlagSupplierImpl`

**Review by abidhkhan — CHANGES_REQUESTED** [review]

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationService.java (hunk 1 above)
Comment (ketanmulay-appdirect) [reply src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationService.java]: Fixed

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationSupplier.java (hunk 2 above)
Comment (ketanmulay-appdirect) [reply src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationSupplier.java]: Fixed

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationSupplierImpl.java (hunk 3 above)
Comment (ketanmulay-appdirect) [reply src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationSupplierImpl.java]: Fixed

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2FeatureFlagSupplier.java (hunk 4 above)
Comment (ketanmulay-appdirect) [reply src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2FeatureFlagSupplier.java]: Fixed

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2FeatureFlagService.java (hunk 5 above)
Comment (ketanmulay-appdirect) [reply src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2FeatureFlagService.java]: Fixed

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2FeatureFlagSupplierImpl.java (hunk 6 above)
Comment (ketanmulay-appdirect) [reply src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2FeatureFlagSupplierImpl.java]: Fixed

File: src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationService.java (hunk 1 above)
Comment (ketanmulay-appdirect) [reply src/main/java/com/appdirect/sdk/web/oauth/DeveloperSpecificOAuth2AuthorizationService.java]: Fixed

**Review by nsadaphal — DISMISSED** [review]
LGTM

**Review by reshmakadam-appdirect — DISMISSED** [review]

**Review by rajbhupendra588 — DISMISSED** [review]

**Review by adityaraj-appdirect — APPROVED** [review]

**Review by abidhkhan — DISMISSED** [review]

**Review by adityaraj-appdirect — APPROVED** [review]

**Review by ssanghavi-appdirect — APPROVED** [review]
LGTM

**Pull request merged**